migrations/20200515235702-initial-table-creation.js

```diff
@@ -0,0 +1,66 @@
+
+module.exports = {
+  up: async (queryInterface, Sequelize) => {
+    /*
+      Add altering commands here.
+      Return a promise to correctly handle asynchronicity.
+
+      Example:
+      return queryInterface.createTable('users', { id: Sequelize.INTEGER });
+    */
+    return await queryInterface.createTable('authors', {
+      id: { type: Sequelize.INTEGER, autoIncrement: true, primaryKey: true },
+      nameFirst: { type: Sequelize.STRING, allowNull: false },
+      nameLast: { type: Sequelize.STRING, allowNull: false },
+      createdAt: { type: Sequelize.DATE, defaultValue: Sequelize.literal('CURRENT_TIMESTAMP') },
+      updatedAt: {
+        type: Sequelize.DATE,
+        defaultValue: Sequelize.literal('CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP')
+      },
+      deletedAt: { type: Sequelize.DATE },
+
+    }), await queryInterface.createTable('genres', {
+      id: { type: Sequelize.INTEGER, autoIncrement: true, primaryKey: true },
+      name: { type: Sequelize.STRING, allowNull: false },
+      createdAt: { type: Sequelize.DATE, defaultValue: Sequelize.literal('CURRENT_TIMESTAMP') },
+      updatedAt: {
+        type: Sequelize.DATE,
+        defaultValue: Sequelize.literal('CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP')
+      },
+      deletedAt: { type: Sequelize.DATE },
+
+    }), await queryInterface.createTable('novels', {
+      id: { type: Sequelize.INTEGER, autoIncrement: true, primaryKey: true },
+      authorId: { type: Sequelize.INTEGER, references: { model: 'authors', key: 'id' } },
+      title: { type: Sequelize.STRING, allowNull: false },
+      createdAt: { type: Sequelize.DATE, defaultValue: Sequelize.literal('CURRENT_TIMESTAMP') },
+      updatedAt: {
+        type: Sequelize.DATE,
+        defaultValue: Sequelize.literal('CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP')
+      },
+      deletedAt: { type: Sequelize.DATE },
+
+    }), await queryInterface.createTable('novelGenres', {
+      genreId: { type: Sequelize.INTEGER, primarykey: true, references: { model: 'genres', key: 'id' } },
+      novelId: { type: Sequelize.INTEGER, primarykey: true, references: { model: 'novels', key: 'id' } },
+      createdAt: { type: Sequelize.DATE, defaultValue: Sequelize.literal('CURRENT_TIMESTAMP') },
+      updatedAt: {
+        type: Sequelize.DATE,
+        defaultValue: Sequelize.literal('CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP')
+      },
+      deletedAt: { type: Sequelize.DATE },
+    })
+  },
+
+  down: (queryInterface) => {
+    /*
+      Add reverting commands here.
+      Return a promise to correctly handle asynchronicity.
+  
+      Example:
+      return queryInterface.dropTable('users');
+    */
+
+    return queryInterface.dropTable('novels'), queryInterface.dropTable('authors'), queryInterface.dropTable('genres'), queryInterface.dropTable('novelGenres')
```

Comment:
Without the awaiting here all of these run at the same time meaning it could attempt to drop novels before dropping novelGenres which would cause it to fail.

```JavaScript
await queryInterface.dropTable('novelGenres')
await queryInterface.dropTable('novels')
await queryInterface.dropTable('authors')

return queryInterface.dropTable('genres') 
```